pkg/config/config.go

```diff
@@ -981,6 +981,18 @@ func getMultipleEndpointsWithConfig(config Config) (map[string][]string, error)
 	return keysPerDomain, nil
 }
 
+// ComputeAPIDomain returns a domain to be used for API validation
+func ComputeAPIDomain(domain string) string {
+	apiDomain := ""
+	matcher := strings.Replace(domain, "https://", "", 1)
```

Comment:
this function takes a valid URL as argument, so I think we should parse the URL instead of using `string.Replace`, similar to here: https://github.com/DataDog/datadog-agent/pull/4958/files#diff-091ace02b49a46f55f2586cfec89a160R885